framework/include/loops/ComputeFVFluxThread.h

```diff
@@ -222,7 +222,7 @@ ThreadedFaceLoop<RangeType>::operator()(const RangeType & range, bool bypass_thr
     }
     catch (libMesh::LogicError & e)
     {
-      throw MooseException("We caught a libMesh error");
+      throw MooseException("We caught a libMesh error: " + std::string(e.what()));
```

Comment:
I learned a word for this from @roystgnr : "Atavism" :-D

Reply:
I don't understand ... does this actually end up throwing an exception? If so, I don't see how

Reply:
I can verify that it does indeed throw ... this is illustrating that I have some missing knowledge around exceptions. How does it throw without the `throw` specifier?

Reply:
just like any function can throw without a throw specifier...?

Reply:
By the way, this is the first time I hear about the throw specifier in a context other than the unconditional `throw()` or the `noexcept` operator(?).

https://devblogs.microsoft.com/oldnewthing/20180928-00/?p=99855

Reply:
> just like any function can throw without a throw specifier...?

You're taking for granted that I know that. I don't haha. I thought you only threw exceptions by doing `throw foo`. But obviously that is not the case. So I want to understand how we're throwing an exception without an explicit `throw`

Reply:
Hang on, I thought you were talking about the "throw specifier", and as far as I know that's part of the function declaration. It informs the compiler that a function could potentially throw (and pre C++11 what it would throw), or in case of the `nothrow` keyword it would guarantee that no throwing occurs. I think the `throw` specifier is optional and the `throw(type list)` version is even deprecated.

The throw keyword that actually throws the exception is a different story. Of course that's needed to throw an exception, but here that happens _inside_ the `mooseException` function.

Reply:
Oh em gee I missed the fact that we changed the capitalization from `MooseException` to `mooseException`. That's what caused my confusion